engine/src/main/java/org/terasology/engine/TerasologyEngine.java

```diff
@@ -131,11 +128,14 @@
     private GameState pendingState;
     private Set<StateChangeSubscriber> stateChangeSubscribers = Sets.newLinkedHashSet();
 
-    private enum EngineState {UNINITIALIZED, INITIALIZED, RUNNING, DISPOSED}
+    private EngineStatus status = StandardGameStatus.UNSTARTED;
+    private final List<EngineStatusSubscriber> statusSubscriberList = new CopyOnWriteArrayList<>();
 
-    private EngineState engineState = EngineState.UNINITIALIZED;
-
-    private final TaskMaster<Task> commonThreadPool = TaskMaster.createFIFOTaskMaster("common", MAX_NUMBER_THREADS);
+    private transient boolean initialised;
+    private transient boolean shutdownRequested;
+    private transient boolean shuttingDown;
+    private transient boolean running;
+    private transient boolean disposed;
```

Comment:
This block of transient booleans is interesting.

 First I wonder why they have been made transient. I seem to understand this is to avoid them being serialized. But is the engine ever serialized?

Second, they all seem to be mutually exclusive. Which is what led me months ago to make an enum out of their predecessors. Why the revert to separate booleans then?